diff --git a/multirom.cpp b/multirom.cpp
index a186ce9..da254c2 100644
--- a/multirom.cpp
+++ b/multirom.cpp
@@ -1673,7 +1673,7 @@ bool MultiROM::extractBootForROM(std::string base)
 
 	// copy needed files
 	static const char *cp_f[] = {
-		"*.rc", "default.prop", "init", "main_init", "fstab.*",
+		"*.rc", "default.prop", "init", "main_init", "fstab.*", "*.sh",
 		// Since Android 4.3 - for SELinux
 		"file_contexts", "property_contexts", "seapp_contexts", "sepolicy",
 		NULL
