diff --git a/mrominstaller.cpp b/mrominstaller.cpp
index 03dd33b..4cbbf8c 100644
--- a/mrominstaller.cpp
+++ b/mrominstaller.cpp
@@ -19,7 +19,7 @@ extern "C" {
 #include "twcommon.h"
 }
 
-#define INTERNAL_MEM_LOC_TXT "Internal memory"
+#define INTERNAL_MEM_LOC_TXT "/data"
 
 MROMInstaller::MROMInstaller()
 {
diff --git a/multirom.cpp b/multirom.cpp
index db0eb4f..398e331 100644
--- a/multirom.cpp
+++ b/multirom.cpp
@@ -194,7 +194,8 @@ std::string MultiROM::listInstallLocations()
 
 	while((dt = readdir(d)))
 	{
-		if(strncmp(dt->d_name, "mmcblk0", 7) == 0 || strncmp(dt->d_name, "dm-", 3) == 0)
+//		if( strncmp(dt->d_name, "mmcblk0", 7) == 0                                                || strncmp(dt->d_name, "dm-", 3) == 0)
+		if((strncmp(dt->d_name, "mmcblk0", 7) == 0 && strncmp(dt->d_name, "mmcblk0p27", 10) != 0) || strncmp(dt->d_name, "dm-", 3) == 0)
 			continue;
 		snprintf(path, sizeof(path), "/dev/block/%s", dt->d_name);
 
@@ -1619,7 +1620,7 @@ bool MultiROM::extractBootForROM(std::string base)
 
 	// copy needed files
 	static const char *cp_f[] = {
-		"*.rc", "default.prop", "init", "main_init", "fstab.*",
+		"*.rc", "default.prop", "init", "main_init", "fstab.*", "*.sh",
 		// Since Android 4.3 - for SELinux
 		"file_contexts", "property_contexts", "seapp_contexts", "sepolicy",
 		NULL
diff --git a/multirom.h b/multirom.h
index e6ac98c..ab2f3d8 100644
--- a/multirom.h
+++ b/multirom.h
@@ -52,7 +52,7 @@ enum
 #define INTERNAL_NAME "Internal"
 #define REALDATA "/realdata"
 #define MAX_ROM_NAME 26
-#define INTERNAL_MEM_LOC_TXT "Internal memory"
+#define INTERNAL_MEM_LOC_TXT "/data"
 
 // Not defined in android includes?
 #define MS_RELATIME (1<<21)
