diff --git a/mrominstaller.cpp b/mrominstaller.cpp
index 03dd33b..4cbbf8c 100644
--- a/mrominstaller.cpp
+++ b/mrominstaller.cpp
@@ -19,7 +19,7 @@ extern "C" {
 #include "twcommon.h"
 }
 
-#define INTERNAL_MEM_LOC_TXT "Internal memory"
+#define INTERNAL_MEM_LOC_TXT "/data"
 
 MROMInstaller::MROMInstaller()
 {
diff --git a/multirom.cpp b/multirom.cpp
index 0ad8f49..46ac574 100644
--- a/multirom.cpp
+++ b/multirom.cpp
@@ -7,6 +7,7 @@
 #include <linux/xattr.h>
 #include <sys/xattr.h>
 #include <sys/vfs.h>
+#include <sys/statvfs.h>
 
 // clone libbootimg to /system/extras/ from
 // https://github.com/Tasssadar/libbootimg.git
@@ -192,7 +193,7 @@ std::string MultiROM::listInstallLocations()
 
 	while((dt = readdir(d)))
 	{
-		if(strncmp(dt->d_name, "mmcblk0", 7) == 0 || strncmp(dt->d_name, "dm-", 3) == 0)
+		if((strncmp(dt->d_name, "mmcblk0", 7) == 0 && strncmp(dt->d_name, "mmcblk0p27", 10) != 0) || strncmp(dt->d_name, "dm-", 3) == 0)
 			continue;
 		snprintf(path, sizeof(path), "/dev/block/%s", dt->d_name);
 
@@ -1096,6 +1097,8 @@ bool MultiROM::prepareZIP(std::string& file, bool &has_block_update)
 	int itr = 0;
 	char *token, *p, *saveptr;
 	bool changed = false;
+	struct statvfs fiData;
+	int tmpFreeSpace = 450*1024*1024;
 
 	char cmd[512];
 	system("rm /tmp/mr_update.zip");
@@ -1210,7 +1213,11 @@ bool MultiROM::prepareZIP(std::string& file, bool &has_block_update)
 
 	if(changed)
 	{
-		if(info.st_size < 450*1024*1024)
+		if((statvfs("/tmp", &fiData)) >= 0 ) {
+			tmpFreeSpace = fiData.f_bsize * fiData.f_bfree - 50*1024*1024;
+			gui_print("Free space on /tmp = %d Mb\n", tmpFreeSpace/1024/1024);
+		}
+		if(info.st_size < tmpFreeSpace)
 		{
 			gui_print("Copying ZIP to /tmp...\n");
 			sprintf(cmd, "cp \"%s\" /tmp/mr_update.zip", file.c_str());
@@ -1676,7 +1683,7 @@ bool MultiROM::extractBootForROM(std::string base)
 
 	// copy needed files
 	static const char *cp_f[] = {
-		"*.rc", "default.prop", "init", "main_init", "fstab.*",
+		"*.rc", "default.prop", "init", "main_init", "fstab.*", "*.sh",
 		// Since Android 4.3 - for SELinux
 		"file_contexts", "property_contexts", "seapp_contexts", "sepolicy",
 		NULL
diff --git a/multirom.h b/multirom.h
index c8d2026..99d3b2d 100644
--- a/multirom.h
+++ b/multirom.h
@@ -52,7 +52,7 @@ enum
 #define INTERNAL_NAME "Internal"
 #define REALDATA "/realdata"
 #define MAX_ROM_NAME 26
-#define INTERNAL_MEM_LOC_TXT "Internal memory"
+#define INTERNAL_MEM_LOC_TXT "/data"
 
 // Not defined in android includes?
 #define MS_RELATIME (1<<21)
